contracts/Aavegotchi/facets/ItemsRolesRegistryFacet.sol

```diff
@@ -0,0 +1,251 @@
+            require(
+                s.itemsDeposits[_grantRoleData.nonce].tokenAddress == _grantRoleData.tokenAddress,
+                "ItemsRolesRegistryFacet: tokenAddress mismatch"
+            );
+            require(s.itemsDeposits[_grantRoleData.nonce].tokenId == _grantRoleData.tokenId, "ItemsRolesRegistryFacet: tokenId mismatch");
+            require(s.itemsDeposits[_grantRoleData.nonce].tokenAmount == _grantRoleData.tokenAmount, "ItemsRolesRegistryFacet: tokenAmount mismatch");
+
+            RoleData storage _roleData = s.itemsRoleAssignments[_grantRoleData.nonce];
+            require(
+                _roleData.expirationDate < block.timestamp || _roleData.revocable,
+                "ItemsRolesRegistryFacet: depositId is not expired or is not revocable"
+            );
+        }
+        _grantOrUpdateRole(_grantRoleData);
+    }
+
+    function _grantOrUpdateRole(RoleAssignment memory _grantRoleData) internal {
+        s.itemsRoleAssignments[_grantRoleData.nonce] = RoleData(
+            _grantRoleData.grantee,
+            _grantRoleData.expirationDate,
+            _grantRoleData.revocable,
+            _grantRoleData.data
+        );
+
+        emit RoleGranted(
+            _grantRoleData.nonce,
+            UNIQUE_ROLE,
+            _grantRoleData.tokenAddress,
+            _grantRoleData.tokenId,
+            _grantRoleData.tokenAmount,
+            _grantRoleData.grantor,
+            _grantRoleData.grantee,
+            _grantRoleData.expirationDate,
+            _grantRoleData.revocable,
+            _grantRoleData.data
+        );
+    }
+
+    function revokeRoleFrom(uint256 _depositId, bytes32 _role, address _grantee) external override validRoleAndGrantee(_role, _grantee, _depositId) {
+        RoleData memory _roleData = s.itemsRoleAssignments[_depositId];
+        DepositInfo memory _depositInfo = s.itemsDeposits[_depositId];
+
+        address caller = _findCaller(_roleData, _depositInfo);
+        if (_roleData.expirationDate > block.timestamp && !_roleData.revocable) {
+            // if role is not expired and is not revocable, only the grantee can revoke it
+            require(caller == _roleData.grantee, "ItemsRolesRegistryFacet: depositId is not expired or is not revocable");
+        }
+
+        uint256 _gotchiId = s.depositIdToItemIdToGotchiId[_depositId][_depositInfo.tokenId];
+        EquipedDelegatedItemInfo memory _equipedDelegatedItemInfo = s.gotchiIdToItemIdToDepositId[_gotchiId][_depositInfo.tokenId];
+        uint256 _balanceToUnequip = _equipedDelegatedItemInfo.balance;
+
+        if (_balanceToUnequip > 0) {
+            _unequipWearable(_gotchiId, _depositInfo.tokenId, _balanceToUnequip);
+
+            delete s.depositIdToItemIdToGotchiId[_depositId][_depositInfo.tokenId];
+            delete s.gotchiIdToItemIdToDepositId[_gotchiId][_depositInfo.tokenId];
+        }
```

Comment:
Move deletes to unequip function, and rename function to `_unequipDelegatedWearable`